### ISSUE TITLE
[css-transforms-2] definition of Accumulated 3D Transformation Matrix has incorrect loop exit condition

### ISSUE THREAD

**Issue by dbaron** [w3c/csswg-drafts#6191]
I've read it a few times, and I think that the [definition of Accumulated 3D Transformation Matrix](https://drafts.csswg.org/css-transforms-2/#accumulated-3d-transformation-matrix-computation) has the exit condition of the loop (step 4) stated incorrectly.  I've read it a bunch of times... but I'd really like some other folks (cc @mattwoodrow @smfr) to double-check this, though!

It currently says:

> While *current element* is not the element that establishes the transformed element’s 3D rendering context:

However, I believe the loop should execute when *current element* **is** the element that establishes the 3D rendering context, and then the loop should terminate *after* that iteration.  In other words, I think this should instead say:

> While *current element* is an element in the transformed element's 3D rendering context:

I've written a [testcase](https://dbaron.org/css/test/2021/what-establishes-3drc) that is interoperable across Chrome, Firefox, and Safari, and that I think matches what I'm proposing.

But, again, I'd really like others to confirm that I'm not misreading/misthinking this somehow!

**Comment by mattwoodrow** [comment]
I agree, we definitely want to include the transform of the establishing element.

I think there's been some confusion and changes around this in the past, where sometimes an element with `transform-style:flat` was considered to the element that establishes the 3D rendering context.

**Comment by smfr** [comment]
I think the intent was that the "Accumulated 3D Transformation Matrix" was the matrix you use when rendering the element into the flattening plane of the "establishing" element. So in that sense it would exclude the transform of the establishing element.
